File: frontend/ProjectManagement.Core/Models/CreateProjectRequest.cs
namespace ProjectManagement.Core.Models;

/// <summary>
/// Request to create a new project.
/// </summary>
public sealed record CreateProjectRequest
{
    /// <summary>
    /// Project title (1-200 characters).
    /// </summary>
    public required string Title { get; init; }

    /// <summary>
    /// Optional description (max 10000 characters).
    /// </summary>
    public string? Description { get; init; }

    /// <summary>
    /// Unique project key (2-20 alphanumeric characters).
    /// Will be stored uppercase.
    /// </summary>
    public required string Key { get; init; }
}

File: backend/pm-server/src/api/projects/create_project_request.rs
use serde::Deserialize;

#[derive(Debug, Deserialize)]
pub struct CreateProjectRequest {
    /// Project title (required)
    pub title: String,

    /// Unique short key, e.g., "PROJ" (required)
    pub key: String,

    /// Optional description
    #[serde(default)]
    pub description: Option<String>,
}
